lib/publiccloud/azure.pm

```diff
@@ -385,12 +399,13 @@ sub create_image_definition {
     record_info("img-defs", "Existing image definitions:\n$images");
 
     my $sku = get_var("PUBLIC_CLOUD_AZURE_SKU", $default_sku);
+
     my $gen = ($sku =~ "gen2" ? "V2" : "V1");
     my $tags = generate_tags();
 
-    my $arch = (get_var('PUBLIC_CLOUD_ARCH', get_required_var('ARCH'))) eq 'x86_64' ? 'x64' : 'Arm64';
+    my $arch = az_arch();
     my $publisher = get_var("PUBLIC_CLOUD_AZURE_PUBLISHER", "qe-c");
-    my $offer = get_var("PUBLIC_CLOUD_AZURE_OFFER", get_var('DISTRI') . '-' . get_var('VERSION') . '-' . get_var('FLAVOR') . '-' . $arch);
+    my $offer = get_var("PUBLIC_CLOUD_AZURE_OFFER", $self->generate_azure_image_definition(1));
```

Comment:
Note that using here `$basename: 1`, the expected _default name_ of the old version is composed even when `PUBLIC_CLOUD_AZURE_IMAGE_DEFINITION` is defined.

Reply:
The name of this variable does not make sense to me.

Reply:
Just don't define `PUBLIC_CLOUD_AZURE_IMAGE_DEFINITION` and you don't need this workaround.

Reply:
it was a suggestion in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/19181#discussion_r1583645149 that I accepted, being it self-explaining from the routine context

Reply:
> Just don't define `PUBLIC_CLOUD_AZURE_IMAGE_DEFINITION` and you don't need this workaround.

Why ? the `PUBLIC_CLOUD_AZURE_IMAGE_DEFINITION` was there since long time I cannot remove it.

Reply:
> Just don't define `PUBLIC_CLOUD_AZURE_IMAGE_DEFINITION` and you don't need this workaround.

The `PUBLIC_CLOUD_AZURE_IMAGE_DEFINITION` option is there since long time, meaning: if `someone defines that name here` (without check of the name syntax), keep that **name** and skip the full routine. But in that case, the call in the _new default_ above would have that **name** that is Not what original code expected.
Therefore, the `forced` name composition needs that exception.